### ISSUE TITLE
Notes on 2.0

### ISSUE THREAD

**Issue by herredro** [herredro/dashboard_react#1]
## 1. Unclear Code features
https://github.com/herredro/dashboard_react/blob/ef43bca6d982b0b335d92cfb483516efc34696b8/src/MapContainer/index.js#L19
- [ ] What is the var *classes* for?
  - So far, there is no *className*-prop passed to *MapContainer* to push in list (-> undefined)
  - Stuff I could do with it
    - [ ] Button coloring depending on state (right, @olivierbrcknr?)
    - [ ] ... 

https://github.com/herredro/dashboard_react/blob/ef43bca6d982b0b335d92cfb483516efc34696b8/src/Dash.js#L14-L16
 - [ ] where is it called? Is it redundant in this case @olivierbrcknr ?

https://github.com/herredro/dashboard_react/blob/ef43bca6d982b0b335d92cfb483516efc34696b8/src/Dash.js#L40-L42
- [ ] callBack - when is this event triggered and where?
  I only see *setFocus* being called directly (in the callback function of *DashNav*).
 --> 

## 2. Unclear React features

https://github.com/herredro/dashboard_react/blob/ef43bca6d982b0b335d92cfb483516efc34696b8/src/DashNav/index.js#L25
- [ ] @olivierbrcknr Is this "the way to do it"? I.e.: is it generaly better style to call a local function first, that then calls the prop function?
  > I would have done "onClick={props.clickHandler}" 
    (@olivierbrcknr, would this ^ be called a callback or just a function that was passed as a prop?)

**Comment by olivierbrcknr** [comment]
From top to bottom:

1.  the `prop.className` variable is there just in case, if you want to add classes in the upper level, that it  would be added there as well. Always make components that they can standalone and are prepared for eventualities
2.  Yes you are right, messed this one up, you can delete that function
3. CallBack is a prop that I created and is passed to the component → that's also why "Unclear React features" is there, because here the function triggers the callBack function
https://github.com/herredro/dashboard_react/blob/ef43bca6d982b0b335d92cfb483516efc34696b8/src/DashNav/index.js#L10-L15
This way you can push states up the tree to the parent

**Comment by herredro** [comment]
Another one...
https://github.com/herredro/dashboard_react/blob/ef43bca6d982b0b335d92cfb483516efc34696b8/src/MapContainer/index.js#L24-L27
...is this actually doing anything if useEffect() is empty?

**Comment by olivierbrcknr** [comment]
Nope, it's not. I just left it there for you to play around, especially with the map / leaflet component
